packages/@aws-cdk/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -62,12 +62,38 @@ async function createLogGroupSafe(logGroupName: string, region?: string, options
  * @param retentionInDays the number of days to retain the log events in the specified log group.
  */
 async function setRetentionPolicy(logGroupName: string, region?: string, options?: SdkRetryOptions, retentionInDays?: number) {
-  const cloudwatchlogs = new AWS.CloudWatchLogs({ apiVersion: '2014-03-28', region, ...options });
-  if (!retentionInDays) {
-    await cloudwatchlogs.deleteRetentionPolicy({ logGroupName }).promise();
-  } else {
-    await cloudwatchlogs.putRetentionPolicy({ logGroupName, retentionInDays }).promise();
-  }
+  // The same as in createLogGroupSafe(), here we could end up with the race
+  // condition where a log group is either already being created or its retention
+  // policy is being updated. This would result in an OperationAbortedException,
+  // which we will try to catch and retry the command a number of times before failing
+  let retryCount = options?.maxRetries == undefined ? 10 : options.maxRetries;
+  const delay = options?.retryOptions?.base == undefined ? 10 : options.retryOptions.base;
+  do {
+    try {
+      const cloudwatchlogs = new AWS.CloudWatchLogs({ apiVersion: '2014-03-28', region, ...options });
+      if (!retentionInDays) {
+        await cloudwatchlogs.deleteRetentionPolicy({ logGroupName }).promise();
+      } else {
+        await cloudwatchlogs.putRetentionPolicy({ logGroupName, retentionInDays }).promise();
+      }
+      return;
+
+    } catch (error) {
+      if (error.code === 'OperationAbortedException') {
+        if (retryCount > 0) {
+          retryCount--;
+          await new Promise(resolve => setTimeout(resolve, delay));
+          continue;
+        } else {
+          // The log group is still being created by another execution but we are out of retries
+          throw new Error('Out of attempts to create a logGroup');
+        }
+      }
+      // Any other error
+      console.error(error);
```

Comment:
I know `createLogGroupSafe()` does this as well, but let's remove these `console` statements.

Reply:
Should I remove them from `createLogGroupSafe()` as well?

Reply:
I removed it from both places, since the error gets caught and logged in the calling method.